### PR TITLE
add checkbox

### PR DESCRIPTION
**Pull request by SimonDanisch**

### CODE REVIEW

**Comment by SimonDanisch** [comment]
@piever you may want to take a look at this, since its finally doing your much wanted unset attribute ;)
@yakir12  hope this helps ;)

File: src/hyperscript_integration.jl [hunk 1]
```diff
@@ -42,12 +42,16 @@ using .DOM
...
 attribute_render(session, parent, attribute, x) = string(x)
+attribute_render(session, parent, attribute, x::Nothing) = x
 function attribute_render(session, parent, attribute, obs::Observable)
     onjs(session, obs, js"""
     function (value){
         var node = $(parent);
         if(node){
-            if(node[$attribute] != value){
+            // we encode nothing == null to mean attribute not set!
+            if(value == null){
```
Comment (piever) [inline src/hyperscript_integration.jl]: This is a bit in contradiction with the way [HyperScript.jl](https://github.com/yurivish/Hyperscript.jl) works, AFAIU. From the last line in the docs:
> Using `nothing` as the value of a `DOM` attribute creates a valueless attribute, e.g. `<input checked />`.
If we have to add it "by hand" I wonder whether we are better off using `false` for this, or maybe a custom `Unset` type.

Alternatively, as WebIO allows to "unset" attributes with `nothing`, we could check with Hyperscript if they are open to changing that behavior.

Comment (SimonDanisch) [reply src/hyperscript_integration.jl]: Ah! I guess I see what's going on here... I already messed with the attribute rendering, and made it ignore nothing (by rendering it to a string before giving it to hyperscript) :D

Comment (SimonDanisch) [reply src/hyperscript_integration.jl]: I guess we need all those cases right?
* valueless attribute
* not set
* set to value
Maybe nothing == valueless, missing = removed?

Comment (piever) [reply src/hyperscript_integration.jl]: Good idea! The custom type is probably overkill. So actually the proper way to generate clean HTML would be to use `Observable{Union{Nothing, Missing}}` (AFAIU, valueless is the good HTML practice for boolean attributes).

File: src/widgets.jl [hunk 2]
```diff
@@ -155,3 +155,29 @@ function jsrender(session::Session, slider::RangeSlider)
...
+    value::Observable{Bool}
+    attributes::Dict{Symbol, Any}
+end
+
+function Checkbox(value::Bool; kw...)
+    return Checkbox(Observable(value), Dict{Symbol, Any}(kw))
+end
+
+function JSServe.jsrender(tb::Checkbox)
+    # unchecked in css is encoded by leaving out the attribute (*sigh*)
```
Comment (piever) [inline src/widgets.jl]: I agree with the "sigh", but this is actually an HTML thing, and HTML5 adds their own bizarre restrictions to try and make things less counterintuitive, see [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes#Boolean_Attributes).

**Comment by SimonDanisch** [comment]
This fixes https://github.com/SimonDanisch/JSServe.jl/issues/27

**Comment by SimonDanisch** [comment]
# [Codecov](https://codecov.io/gh/SimonDanisch/JSServe.jl/pull/35?src=pr&el=h1) Report
> Merging [#35](https://codecov.io/gh/SimonDanisch/JSServe.jl/pull/35?src=pr&el=desc) into [master](https://codecov.io/gh/SimonDanisch/JSServe.jl/commit/8e08afc7f6e90f0ee93e62e24e2d11b32d1b2058&el=desc) will **decrease** coverage by `1.19%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/SimonDanisch/JSServe.jl/pull/35?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master      #35      +/-   ##
==========================================
- Coverage   90.96%   89.77%   -1.20%     
==========================================
  Files          17       17              
  Lines         620      626       +6     
==========================================
- Hits          564      562       -2     
- Misses         56       64       +8     
```


| [Impacted Files](https://codecov.io/gh/SimonDanisch/JSServe.jl/pull/35?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/util.jl](https://codecov.io/gh/SimonDanisch/JSServe.jl/pull/35/diff?src=pr&el=tree#diff-c3JjL3V0aWwuamw=) | `86.66% <0.00%> (-6.67%)` | :arrow_down: |
| [src/hyperscript\_integration.jl](https://codecov.io/gh/SimonDanisch/JSServe.jl/pull/35/diff?src=pr&el=tree#diff-c3JjL2h5cGVyc2NyaXB0X2ludGVncmF0aW9uLmps) | `82.05% <0.00%> (-7.61%)` | :arrow_down: |
| [src/types.jl](https://codecov.io/gh/SimonDanisch/JSServe.jl/pull/35/diff?src=pr&el=tree#diff-c3JjL3R5cGVzLmps) | `83.33% <0.00%> (-2.78%)` | :arrow_down: |
| [src/session.jl](https://codecov.io/gh/SimonDanisch/JSServe.jl/pull/35/diff?src=pr&el=tree#diff-c3JjL3Nlc3Npb24uamw=) | `85.55% <0.00%> (-1.12%)` | :arrow_down: |
| [src/js\_source.jl](https://codecov.io/gh/SimonDanisch/JSServe.jl/pull/35/diff?src=pr&el=tree#diff-c3JjL2pzX3NvdXJjZS5qbA==) | `95.23% <0.00%> (-0.42%)` | :arrow_down: |
| [src/dependencies.jl](https://codecov.io/gh/SimonDanisch/JSServe.jl/pull/35/diff?src=pr&el=tree#diff-c3JjL2RlcGVuZGVuY2llcy5qbA==) | `95.23% <0.00%> (-0.22%)` | :arrow_down: |
| [src/display.jl](https://codecov.io/gh/SimonDanisch/JSServe.jl/pull/35/diff?src=pr&el=tree#diff-c3JjL2Rpc3BsYXkuamw=) | `100.00% <0.00%> (ø)` | :arrow_up: |
| [src/jscall.jl](https://codecov.io/gh/SimonDanisch/JSServe.jl/pull/35/diff?src=pr&el=tree#diff-c3JjL2pzY2FsbC5qbA==) | `92.59% <0.00%> (+2.59%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/SimonDanisch/JSServe.jl/pull/35?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/SimonDanisch/JSServe.jl/pull/35?src=pr&el=footer). Last update [8e08afc...f36f3ac](https://codecov.io/gh/SimonDanisch/JSServe.jl/pull/35?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Pull request merged**